release-notes.txt

```diff
@@ -1,3 +1,841 @@
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2022.10 release includes, among many many bug fixes and improvements, the
+following prominent contributions:
+
+### ESP32 family
+
+RIOT has been ported to the ESP32-C3, ESP32-S2, and the ESP32-S3 MCUs. In
+addition driver support for the integrated BLE transceiver of the ESP32 MCUs
+has been added.
+
+### Support for Priority Inheritance in `core_mutex`
+
+If a shared resource guarded by a mutex is used by a high priority and a low
+priority thread, the high priority thread may end up waiting on the low priority
+thread to return the resource. If medium priority threads now interrupt the
+low priority thread while working with the shared resource, the high priority
+thread ends up waiting on both medium and low priority threads. This effect is
+called priority inversion.
+
+Priority inversion can in most cases be solved by carefully assigning priorities
+and resources. In some cases however, this cannot be avoided: Let's consider
+a scenario in which an I2C temperature sensor and an I2C IMU are used. Since
+temperature changes are slow in frequency, the temperature readings are
+assigned the lowest priority, while the IMU readings are assigned the highest
+priority. The sensible approach here is to have both sensors on distinct I2C
+buses. However, if e.g. only a single I2C bus is available by the hardware
+priority inversion cannot be prevented by careful resource and priority
+assignment alone. In these cases the module `core_mutex_priority_inheritance`
+can be used to lend the low priority thread the priority of the high priority
+thread while the high priority thread is waiting for the mutex hold by the
+low priority thread.
+
+After RIOT has already [been launched into space](https://github.com/thingsat)
+it is now ready to for
+[Mars](https://www.cs.unc.edu/~anderson/teach/comp790/papers/mars_pathfinder_long_version.html).
+
+### USB Support for STM32F1 MCUs
+
+While RIOT has longstanding USB support for many STM32 MCU families, the F1
+family was not supported due to some peculiarities in the memory layout. The
+driver has now been extended to handle them correctly. With this, the popular
+STM32F103 bluepill board now supports stdio via USB, which however still
+has to be manually enabled e.g. by:
+
+```sh
+USEMODULE=stdio_cdc_acm BOARD=bluepill make -C examples/default
```

Comment:
It's actually enabled by default for the `bluepill-128kib` versions.
The smaller ones failed to for too many applications with the added size of the USB stack.

Reply:
This is true for `master`. But unless I missed something, making `stdio_cdc_acm` the default never got backported.

Reply:
Ah I didn't realize there was a feature freeze between those two PRs.